### ISSUE TITLE
Mearure of mA

### ISSUE THREAD

**Issue by vespadj** [Upcycle-Electronics/DSN-VC288#1]
Hi!
May be it possible to scale Amperometer to look 5,5 mA at 10 V?
Thanks

**Comment by Upcycle-Electronics** [comment]
Not without rewritting the software from scratch.

**Comment by vespadj** [comment]
I suppose it's true.
In many web store this device is described as range 0-999 mA , 0-9.99 A , people asks how, but any confirms or describes how switch the range, so it has only the 0-9.99 A range.

To take confidence with DSN-VC288 I try to use reset contacts (top left) with a jumper at start up, without others cable.
But now display says:
-60
--0

Probably now it's unusable.

PS: PDF v1.1 and 2.0 seem to have some connection differences for R6 330 Ohm. It need some check in a new thread?

**Comment by Upcycle-Electronics** [comment]
Yes there is a small difference and a problem with the first PDF. I am just too lazy to clone the directory and try to fix it in KiCAD. I don't use the old computer I made this on and have moved on to other projects, sorry.

 I have tried modifying the gain circuit before. I think it fails due to the narrow amplification range of the LM358 when the power rails are zero and 3v3. It is not a rail to rail op amp. It has been a long time since I played with this, but if I recall correctly you should be able to alter the gain up or down by a factor of 10, the limitation preventing this is the op amp. If you swapped the op amp out for a rail to rail version the problem becomes the R6 330R resistor, and it doesn't work either. I think (but am not certain) that this resistor is needed for the startup current requirements of the LM358. This gets it into linear operation. I'm not sure if there is some kind of software fix that helps account for non linearity too. I have built a few current sense amplifiers like this using a LM358 and LM324, but I used a silicon diode on the op amp output, inside the feedback loop. Basically, just put the diode's cathode on the output and treat the anode like the op amp output. (if i have that correct: "P.A.N.I.C." Positive Anode Negative Is Cathode//I'm learning micropython right now so this stuff seems like a distant dream on the back of my mind...) This only works with a single supply rail and ground configuration though. The diode voltage drop forces the op amp into linear operation. I was told this is a classic old-skool hack for this type of circuit, and it worked for me. I almost uploaded one of my projects on this but the board was a single sided design and I learned the hard way: "always make your jumper wires perpendicular to the X and Y planes." The thing was far too ugly to allow anyone to see it, so I don't have a reference here.
-Jake
